### ISSUE TITLE
Infinite refreshing/importing with projects in Dropbox

### ISSUE THREAD

**Issue by jamesellisnyc** [bdkjones/CodeKit-1-Old#610]
I've been experiencing an issue similar to #545 with projects running inside of Dropbox. I have CodeKit 1.6.2. I'm running Dropbox 1.6.18 on OS X 10.8.3.

Steps to replicate:
- Add a Dropbox-managed directory to Codekit as a new project
- Enable project-level settings for this new project

From there the refreshing/linking loop will kick in, maxing out the CPU. The looping only begins when project-level settings are enabled.

Moving the project out of Dropbox eliminates the issue. Pausing/quitting Dropbox has no effect. The issue occurs with any CodeKit project in my Dropbox, it doesn't matter the content of the project. Also, I should note that my Dropbox is in the default location (my home directory).

Thanks, -j

ps. I'm a big fan of CodeKit. Thanks for creating such a wonderful app.

**Comment by bdkjones** [comment]
Hi James,

```
Have you tried resetting the app completely by quitting it and then relaunching it while holding down the shift key? Give that a shot and let me know if you can still replicate the infinite refresh loop with a Dropbox project. Thanks!
```

**Comment by jamesellisnyc** [comment]
@bdkjones Unfortunately I tried that already. I also tried trashing and reinstalling the app as well. Let me know if there's any way I can help you get to the bottom of it.

**Comment by andymcfee** [comment]
I have the same exact problem.  I did the whole quit and hold shift on restart thing and that worked... For a while. And it happened again.  About once or twice a day.  My project is in a Dropbox-managed directory.  (My design buddy is having the same problem).

**Comment by chewwie** [comment]
Bump. I'm also having this issue. Any project folder within Dropbox folder causes an infinite loop and eats up CPU. If I move out of Dropbox folder it works fine. I have tried restarting with the shift key, deleting codekit json file etc but it still goes crazy.

To make sure I wasn't going mad I created a new, empty folder in dropbox then added it as the only project in codekit. As soon as I turn on project specific settings the infinite looping starts.

Please - any suggestions?

**Comment by bdkjones** [comment]
Yea, the culprit here is definitely Dropbox. It's modifying files on disk (syncing them), CodeKit sees that file change and attempts to process accordingly. It changes a file, Dropbox sees that, changes the file, CodeKit sees that, changes the file, as infinitum.

**Comment by chewwie** [comment]
It never used to do this and I've been using codekit and dropbox together for months. Any idea what may have changed? Maybe something in the latest update? Thanks in advance.

**Comment by bdkjones** [comment]
You've definitely tried the shift-reset trick already?

**Comment by thomasplevy** [comment]
This just started occurring for me today -- I've had several projects in dropbox for months with no issues

**Comment by bdkjones** [comment]
Quit app. Relaunch while holding shift key.

**Comment by thomasplevy** [comment]
Thanks Bryan, I tried that with no luck. As soon as I add a project back in the infinite loop starts again.

I tried also clearing the codekit-config file, this fixes the issue, but as soon as I reenable project level settings the infinite loop immediately starts (even with default project level settings).

I moved the project(s) out of dropbox and tried again (with the original project settings) and the project immediately goes away.

I've concluded (as I'm sure you have as well) that the issue is definitely with dropbox)

That said, I'm just taking my projects out of dropbox -- I'm not sure if this is something that you can actually fix and dropbox has been causing other issues for me elsewhere recently.

Thanks,

**Comment by bdkjones** [comment]
Hi Thomas,

```
Dropbox is a known issue. It’s trying to watch and modify files at the same time as CodeKit and bad things happen.
```

**Comment by stef-rausch** [comment]
Just to let you know @bdkjones  : My setup with Dropbox just works fine. Even with per project settings.

**Comment by bdkjones** [comment]
Yep. About 70% of people have no trouble. 30% find all sorts of problems. I have no idea why.
